packages/custom-elements/src/Patch/Document.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import CustomElementInternals from '../CustomElementInternals.js';
+
+import PatchParentNode from './Interface/ParentNode.js';
+import * as Native from './Native.js';
+
+export default function(internals: CustomElementInternals) {
+  Document.prototype.createElement = function(
+                                         this: Document, localName: string) {
+    return internals.createAnElement(this, localName, null);
+  } as typeof Document.prototype.createElement;
```

Comment:
Oh, yup that is pretty intense: https://github.com/microsoft/TypeScript/blob/09441107c1e3f3d807f729b14d7f2dc912d4f8f5/src/lib/dom.generated.d.ts#L4770-L4793

Reply:
On second thought, taking their types as-is seems to work and I'm not sure it makes sense to convert this package to TS and then bail out of the type system because the type is complex.

Reply:
> and then bail out of the type system because the type is complex

The types are there to help us and our users.

Anywhere that they hinder more than help, we should totally just bail out and use a type cast. In some cases the type that most accurately models the code is more complicated than it's worth. I think this is one of those cases. Copying the types from lib.dom.d.ts here wouldn't help users of this package, and I can't see any realistic cases where it would catch any bugs in the CE polyfill.

Reply:
(from offline) I didn't realize that TS doesn't actually check that the function conforms to the overload signatures. So, it doesn't sound like there's any real benefit to adding them here - we might as well cast.